### DISCUSSION TITLE
Getting 0% score for Uni<Response> return type

### DISCUSSION THREAD

**Discussion by omar-zahid** [quarkusio/quarkus#25402] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/57479337/166965637-b344b4bc-aecc-4c7f-83ac-a2b13b37b1d7.png)

I'm using Hibernate reactive panache Active Record method to create API's. However, the writer score is 0% if I return a Uni of type Response. Here's the code.

```
@POST
public Uni<Response> create(Ship ship) {
return Panache.withTransaction(ship::persist)
                .replaceWith(Response.ok().status(CREATED)::build);
    }
```

Any help is appreciated.

Thanks.

**Comment by mschorsch** [comment, marked as answer]
With a Response quarkus is not able to determine the type of the entity and cannot register writers at build time. Instead of Response you should use RestResponse<...> (https://quarkus.io/guides/resteasy-reactive#manually-setting-the-response).

**Reply by omar-zahid** [reply]
Thanks @mschorsch 
After your suggested changes, it works.
![image](https://user-images.githubusercontent.com/57479337/168478718-2faef24b-76f1-478f-9753-8d6e29e95ef5.png)
